features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin/source/src/app/components/ManageAlerts/ManageAlerts.jsx

```diff
@@ -0,0 +1,585 @@
+                + 'for that particular API or Response status code is in between 500 and 600 (By Default)',
+            }),
+        },
+    };
+
+    /**
+     * Checks whether the provided alert is aubscribed.
+     *
+     * @param {number} alertId : The id of the alert.
+     * @return {boolean} True if the alert is subscribed. False otherwise.
+     * */
+    const isAlertSubscribed = (alertId) => {
+        return subscribedAlerts.some((alert) => { return alert.id === alertId; });
+    };
+
+    /**
+     * Handles the alert type select event.
+     *
+     * @param {obj} alert : The selected alert.
+     * */
+    const handleCheckAlert = (alert) => {
+        const alertId = alert.id;
+        let tmpSubscribedAlerts = [...subscribedAlerts];
+        if (isAlertSubscribed(alertId)) {
+            tmpSubscribedAlerts = tmpSubscribedAlerts.filter((sub) => {
+                return sub.id !== alertId;
+            });
+        } else {
+            const newAlert = { id: alertId, name: alertIdMapping[alertId].displayName };
+            tmpSubscribedAlerts.push(newAlert);
+        }
+        setSubscribedAlerts(tmpSubscribedAlerts);
+    };
+
+    const validateEmail = (value) => {
+        const schema = Joi.string().email().empty();
+        const validationError = schema.validate(value).error;
+
+        if (validationError) {
+            const errorType = validationError.details[0].type;
+            if (errorType === 'string.email') {
+                return 'Invalid Email: ' + value;
+            }
+        }
+        return false;
+    };
+
+    const validateEmailList = (values) => {
+        for (const email in values) {
+            if (validateEmail(values[email])) {
+                return validateEmail(values[email]);
+            }
+        }
+        return false;
+    };
+
+    /**
+     * Handles the add email event.
+     *
+     * @param {string} email The email address that is being added.
+     * */
+    const handleAddEmail = (email) => {
+        setEmailsList(email);
+    };
+
+    /**
+     * Handles the email deletion event.
+     *
+     * @param {string} email : The email that is being deleted.
+     * */
+    const handleEmailDeletion = (email) => {
+        const newEmails = emails.filter((oldEmail) => {
+            return oldEmail !== email;
+        });
+        setEmailsList(newEmails);
+    };
+
+    useEffect(() => {
+        const supportedAlertsPromise = api.getSupportedAlertTypes();
+        const subscribedAlertsPromise = api.getSubscribedAlertTypesByUser();
+        Promise.all([supportedAlertsPromise, subscribedAlertsPromise]).then((response) => {
+            if (response[0].status === 204 || response[1].status === 204) {
+                setAnalyticsEnabled(false);
+            } else {
+                setAnalyticsEnabled(true);
+                setSubscribedAlerts(response[1].body.alerts);
```

Comment:
Shall we first destructure these bodies and use them
i:e 
```javascript
[supportedAlerts, subscribedAlerts] = response;
```

Reply:
Fixed with commit: https://github.com/wso2/carbon-apimgt/pull/8662/commits/3dde61d37eba0356fd044756c4ccc61a48686a13